src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx

```diff
@@ -1,128 +1,288 @@
+const getButtonStyles = (kind: ButtonKind, theme: ButtonTheme, disabled: boolean) => {
+  if (kind === 'secondary') return '';
+  if (disabled) return 'dark:bg-volcanic-600';
+
+  if (kind === 'outline') {
+    return cn('border', {
+      'dark:border-evolved-green-700 dark:group-hover:border-evolved-green-500':
+        theme === 'evolved-green',
+      'dark:border-blue-500 dark:group-hover:border-blue-400': theme === 'acrylic-blue',
+      'dark:border-coral-700 dark:group-hover:border-coral-600': theme === 'coral',
+      'dark:border-quartz-500 dark:group-hover:border-quartz-400': theme === 'quartz',
+      'dark:border-mushroom-500 dark:group-hover:border-mushroom-400': theme === 'mushroom-marble',
+    });
+  } else {
+    return cn({
+      'dark:bg-evolved-green-700 dark:group-hover:bg-evolved-green-500': theme === 'evolved-green',
+      'dark:bg-blue-500 dark:group-hover:bg-blue-400': theme === 'acrylic-blue',
+      'dark:fill-coral-700 dark:bg-coral-700 dark:group-hover:bg-coral-600': theme === 'coral',
+      'dark:bg-quartz-500 dark:group-hover:bg-quartz-400': theme === 'quartz',
+      'dark:bg-mushroom-500 dark:group-hover:bg-mushroom-400': theme === 'mushroom-marble',
     });
+  }
+};
 
-/**
- * Component that renders a `CellButton` or a `MinimalButton`.
- */
-export const Button: React.FC<Props> = (props) => {
-  const { kind = 'primary', label, theme, ...buttonProps } = props;
-  const isLabelString = typeof label === 'string';
-
-  if (kind === 'secondary') {
-    const { size, ...minimalButtonProps } = buttonProps;
-    const wrappedLabel = isLabelString ? (
-      <Text as="span" styleAs={size === 'lg' ? 'p-lg' : 'p'}>
-        {label}
-      </Text>
+const getCellStyles = (theme: ButtonTheme, disabled: boolean) => {
+  if (disabled) return 'dark:fill-volcanic-600';
+
+  return cn({
+    'dark:fill-evolved-green-700 dark:group-hover:fill-evolved-green-500':
+      theme === 'evolved-green',
+    'dark:fill-blue-500 dark:group-hover:fill-blue-400': theme === 'acrylic-blue',
+    'dark:fill-coral-700 dark:group-hover:fill-coral-600': theme === 'coral',
+    'dark:fill-quartz-500 dark:group-hover:fill-quartz-400': theme === 'quartz',
+    'dark:fill-mushroom-500 dark:group-hover:fill-mushroom-400': theme === 'mushroom-marble',
+  });
+};
+
+export type ButtonProps = {
+  id?: string;
+  kind?: ButtonKind;
+  theme?: ButtonTheme;
+  label?: React.ReactNode | string;
+  children?: React.ReactNode;
+  icon?: IconName;
+  disabled?: boolean;
+  isLoading?: boolean;
+  className?: string;
+  iconPosition?: 'start' | 'end';
+  iconOptions?: {
+    className?: string;
+    kind?: 'default' | 'outline';
+    customIcon?: React.ReactNode;
+  };
+  buttonType?: 'submit' | 'reset' | 'button' | undefined;
+  onClick?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  href?: string;
+  hrefOptions?: {
+    shallow?: boolean;
+    rel?: string;
+    target?: HTMLAttributeAnchorTarget;
+  };
+  animate?: boolean;
```

Comment:
do we still need this? I think we want everything to be animated

Reply:
I think it's good to leave it in - for icon buttons we don't usually want them to animate.